packages/cozy-harvest-lib/src/components/OAuthWindow.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+import { translate } from 'cozy-ui/react/I18n'
+
+import { prepareOAuth, checkOAuthData, terminateOAuth } from '../helpers/oauth'
+import withDomain from './hoc/withDomain'
+import Popup from './Popup'
+
+const OAUTH_POPUP_HEIGHT = 800
+const OAUTH_POPUP_WIDTH = 800
+
+/**
+ * OAuth window is responsible for managing the OAuth process on the client
+ * side, form storing info into local storage, load OAuth window and listen
+ * for data from it.
+ *
+ * The goal of our OAuth Workflow is to retrieve an account id.
+ *
+ * The main part of our OAuth workflow is done by the stack, which redirect
+ * to the OAuth provider authentification endpoint, handle redirection, creates
+ * account and forward the id of the created account.
+ */
+export class OAuthWindow extends PureComponent {
+  constructor(props, context) {
+    super(props, context)
+    this.handleClose = this.handleClose.bind(this)
+    this.handleMessage = this.handleMessage.bind(this)
+
+    const { domain, konnector } = props
+    const { oAuthStateKey, oAuthUrl } = prepareOAuth(domain, konnector)
+    this.state = { oAuthStateKey, oAuthUrl, succeed: false }
```

Comment:
I think this should be done on mount, since it has a side effect, and that the counterpart is done on unmount. Here we can have the case of an instantiated component, storing data in localstorage, never being mounted, so localstorage is never cleared (since there is no unmounting).